identity_iota_core/src/did/iota_did.rs

```diff
@@ -50,6 +46,18 @@ impl IotaDID {
   /// The length of an Alias ID, which is a BLAKE2b-256 hash (32-bytes).
   pub(crate) const TAG_BYTES_LEN: usize = 32;
 
+  #[ref_cast_custom]
+  /// Convert a `CoreDID` reference to an `IotaDID` reference without checking the referenced value.
+  ///  
+  /// # Warning
+  /// This method should only be called on [`CoreDIDs`](CoreDID) that
+  /// are known to satisfy the requirements of the IOTA UTXO specification.  
+  ///
+  /// # Memory safety
+  ///
+  /// The `ref-cast` crate ensures a memory safe implementation.  
+  pub(crate) const fn from_inner_ref_unchecked(did: &CoreDID) -> &Self;
```

Comment:
```suggestion
  /// Convert a `CoreDID` reference to an `IotaDID` reference without checking the referenced value.
  ///  
  /// # Warning
  /// This method should only be called on [`CoreDIDs`](CoreDID) that
  /// are known to satisfy the requirements of the IOTA UTXO specification.  
  ///
  /// # Memory safety
  ///
  /// The `ref-cast` crate ensures a memory safe implementation.  
  #[ref_cast_custom]
  pub(crate) const fn from_inner_ref_unchecked(did: &CoreDID) -> &Self;
```

We typically put comments above macros.